tasks/set_facts.yml

```diff
@@ -0,0 +1,36 @@
+- name: Get correct path for xl if needed
+  shell: which xl
+  register: xen_vman_xl_path
+  when: xen_vman_xl_path is not defined
+  changed_when: False
+
+- set_fact:
+    xen_vman_xl_path: "{{ xen_vman_xl_path.stdout }}"
+  when: xen_vman_xl_path.stdout is defined
+  changed_when: False
+
+- name: Get correct path for iscsiadm if needed
+  shell: which iscsiadm
+  register: xen_vman_iscsiadm_path
+  when: xen_vman_iscsiadm_path is not defined
+  changed_when: False
+
+- set_fact:
+    xen_vman_iscsiadm_path: "{{ xen_vman_iscsiadm_path.stdout }}"
+  when: xen_vman_iscsiadm_path.stdout is defined
+  changed_when: False
+
+- name: Get current running kernel
+  shell: uname -r
+  register: xen_vman_host_kernel
+  changed_when: False
+
+- name: Set vm kernel to current running kernel if not specifed
+  set_fact:
+    xen_vman_kernel: /boot/vmlinuz-{{ xen_vman_host_kernel.stdout }}
```

Comment:
Why don't you just use /vmlinuz?

Reply:
It's a properly bad idea, to use a symlinked kernel or initrd, since it can panic the vm, when the kernel is updated at the host system and the symlink is replaced with another one, than the actual used kernel.